pkg/k8shandler/elasticsearch_helper.go

```diff
@@ -0,0 +1,369 @@
+package k8shandler
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	certLocalPath = "/tmp/"
+)
+
+type esCurlStruct struct {
+	Method       string // use net/http constants https://golang.org/pkg/net/http/#pkg-constants
+	URI          string
+	RequestBody  string
+	StatusCode   int
+	ResponseBody map[string]interface{}
+	Error        error
+}
+
+func parseBool(path string, interfaceMap map[string]interface{}) bool {
```

Comment:
We should consider using a library that already exists to do this and is used in other projects like the oauth-proxy:https://github.com/jcantrill/elasticsearch-proxy/tree/master/vendor/github.com/bitly/go-simplejson  I believe I have also seen this in some of the network and registry operators

Reply:
we're using `encoding/json` already to unmarshal, but this just abstracts needing to type cast and check at every level.